### ISSUE TITLE
a bug with const and loop

### ISSUE THREAD

**Issue by ooovi** [DiffMu/DiffPrivacyInferenceHs#116]
```
typecheck_hs_from_string("function doo(x)
       for _ in 1:10
          x = 100
       end
       x
end")
```
gives type
```(Fun([([NoFun(Num(Int[100.0])) @ s_2] -> NoFun(Num(Int[100.0])))```
but the input `x` does not have to be const 100

**Comment by MxmUrw** [comment]
This is because we simplified the constraints generated by loops. Since code such as above is useless in any case, we should simply disallow it entirely.

**Comment by MxmUrw** [comment]
For the op we now get
```
---------------------------------------------------------------------------
Type:
{
  -  Int[--]
      @ 0.0
  --------------------------
   -> Int[--]

}
---------------------------------------------------------------------------
Constraints:
   - top:

   - others:
```

**Comment by MxmUrw** [comment]
As the `count` example is tracked in #174, and this issue is fixed and has a test, closing.

**Comment by MxmUrw** [comment]
(The fix is due to #201)
